velodyne_pointcloud/include/velodyne_pointcloud/datacontainerbase.h

```diff
@@ -141,53 +146,86 @@ class DataContainerBase
     config_.fixed_frame = fixed_frame;
     config_.target_frame = target_frame;
 
-    config_.transform = fixed_frame.compare(target_frame) != 0;
-    if (config_.transform && !tf_ptr)
+    // only use somewhat resource intensive tf listener when it is necessary
+    if (!fixed_frame.empty() || !target_frame.empty())
```

Comment:
Shouldn't this be `if (!fixed_frame.empty() || (!target_frame.empty() && target_frame != sensor_frame))`? Seems silly to create the buffer and listener if the sensor frame and the target frame are the same and I don't think it's clear anywhere that the target frame should be left empty if you just want the sensor frame to be passed through unaltered.

Reply:
You are right I will add this. I had this already in my mind.

Reply:
You are right I will add this. I had this already in my mind.

Reply:
I addressed this in my latest commit.